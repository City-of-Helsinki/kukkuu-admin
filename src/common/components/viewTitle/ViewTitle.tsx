import React from 'react';
import { CardHeader } from '@material-ui/core';

const ViewTitle = ({
  basePath,
  record,
  source,
}: {
  basePath?: string;
  record?: any;
  source: string;
}) => {
  const title = `${record.translations.FI.name}`
    ? `${record.translations.FI.name}`
    : '';
  return <CardHeader title={title} />;
};
export default ViewTitle;
