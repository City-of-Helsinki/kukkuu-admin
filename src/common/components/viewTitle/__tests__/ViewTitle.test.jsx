import React from 'react';
import { render, screen } from '@testing-library/react';
import { useRecordContext, RecordContextProvider } from 'react-admin';

import ViewTitle from '../ViewTitle';

// Mock the react-admin hooks and components
vi.mock('react-admin', () => ({
  useRecordContext: vi.fn(),
  // eslint-disable-next-line react/prop-types
  RecordContextProvider: ({ children }) => <div>{children}</div>,
}));

describe('ViewTitle', () => {
  it('renders the title from the record context when translations.FI.name exists', () => {
    const mockRecord = {
      translations: {
        FI: {
          name: 'Test Name',
        },
      },
    };
    useRecordContext.mockReturnValue(mockRecord);

    render(<ViewTitle />);

    expect(screen.getByText('Test Name')).toBeInTheDocument();
  });

  it('renders an empty title when translations.FI.name does not exist', () => {
    const mockRecord = {
      translations: {
        FI: {}, // Missing name
      },
    };
    useRecordContext.mockReturnValue(mockRecord);

    const { container } = render(<ViewTitle />);

    expect(container).toMatchSnapshot();
  });

  it('renders an empty title when record context is empty', () => {
    const mockRecord = {};
    useRecordContext.mockReturnValue(mockRecord);
    const { container } = render(<ViewTitle />);

    expect(container).toMatchSnapshot();
  });
});
