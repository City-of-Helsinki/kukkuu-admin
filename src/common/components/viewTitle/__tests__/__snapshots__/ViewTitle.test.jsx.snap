// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ViewTitle > renders an empty title when record context is empty 1`] = `
<div>
  <div
    class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    />
  </div>
</div>
`;

exports[`ViewTitle > renders an empty title when translations.FI.name does not exist 1`] = `
<div>
  <div
    class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    />
  </div>
</div>
`;
